app/src/test/java/com/example/concordia_campus_guide/ModelTests/BuildingTest.java

```diff
@@ -0,0 +1,265 @@
+package com.example.concordia_campus_guide.ModelTests;
+
+import android.text.TextUtils;
+
+import com.example.concordia_campus_guide.Models.Building;
+import com.example.concordia_campus_guide.Models.Coordinates;
+import com.example.concordia_campus_guide.Models.ListOfCoordinates;
+import com.google.android.gms.maps.model.GroundOverlayOptions;
+import com.google.android.gms.maps.model.LatLng;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+public class BuildingTest {
+    private Building defaultBuilding;
+    private Building buildingWithArgs;
+    private Coordinates centerCoordinates = new Coordinates(45.4972685,-73.5789475);
+    private List<String> availableFloors=new ArrayList<String>(Arrays.asList("00", "1", "8", "9"));
+    private float width = 68;
+    private float height = 68;
+    private float bearing = 34;
+    private String campus = "SGW";
+    private String buildingCode = "H";
+    private String Building_Long_Name = "Henry F. Hall Building";
+    private String address = "1455 De Maisonneuve West";
+    private List<String> departments = new ArrayList<String>(Arrays.asList(
+            "Geography, Planning and Environment",
+            "Political Science, Socialogy and Anthropology, Economics",
+            "School of Irish Studies"
+      ));
+    private List<String> services = new ArrayList<String>(Arrays.asList("Welcome Crew Office",
+            "DB Clarke Theatre",
+            "Dean of Students",
+            "Aboriginal Student Resource Centre",
+            "Concordia Student Union",
+            "IT Service Desk",
+            "Security Office",
+            "Student Success Centre",
+            "Mail Services",
+            "Archives",
+            "Career and Planning Services",
+            "Sexual Assault Resource Centre (SARC)"
+    ));
+
+    Coordinates cornerCoordinate1 = new Coordinates(45.497178,-73.57955);
+    Coordinates cornerCoordinate2 = new Coordinates(45.497708,-73.579035);
+    Coordinates cornerCoordinate3 = new Coordinates(45.497385,-73.578332);
```

Comment:
usually assigning values to variables should take place in the init() method. I feel it would be cleaner if you do that. What do u think ?

Reply:
@narrapangan can you take a look please

Reply:
@Hambrsoom the @Before sign will make it so that method is run before every test, which may not be what @narrapangan wanted to do here. Instead, I recommend using the @BeforeClass, so that these variables can be instantiated:

```@BeforeClass
public static void method()
The BeforeClass annotation indicates that the static method to which is attached must be executed once and before all tests in the class. That happens when the test methods share computationally expensive setup (e.g. connect to database).```

Reply:
In this case, the declaration would still go outside of the @BeforeClass method, but the assignment would be in the @BeforeClass method. This way, you still have access to the variables in the other test methods